spec/rubocop/cop/style/string_concatenation_spec.rb

```diff
@@ -293,6 +293,52 @@
     end
   end
 
+  context 'characters for interpolation inside single quotes' do
+    it 'registers an offense for `#{}`' do
+      expect_offense(<<~'RUBY')
+        "foo" + '#{bar}'
+        ^^^^^^^^^^^^^^^^ Prefer string interpolation to string concatenation.
+      RUBY
+
+      expect_correction(<<~'RUBY')
+        "foo\#{bar}"
+      RUBY
+    end
+
+    it 'registers an offense for `#$`' do
+      expect_offense(<<~'RUBY')
+        "foo" + '#$bar'
+        ^^^^^^^^^^^^^^^ Prefer string interpolation to string concatenation.
+      RUBY
+
+      expect_correction(<<~'RUBY')
+        "foo\#$bar"
+      RUBY
+    end
+
+    it 'registers an offense for `#@`' do
+      expect_offense(<<~'RUBY')
+        "foo" + '#@bar'
+        ^^^^^^^^^^^^^^^ Prefer string interpolation to string concatenation.
+      RUBY
+
+      expect_correction(<<~'RUBY')
+        "foo\#@bar"
+      RUBY
+    end
+
+    it 'registers an offense for `#@@`' do
+      expect_offense(<<~'RUBY')
+        "foo" + '#@@bar'
+        ^^^^^^^^^^^^^^^^ Prefer string interpolation to string concatenation.
+      RUBY
+
+      expect_correction(<<~'RUBY')
+        "foo\#@@bar"
+      RUBY
+    end
```

Comment:
Good test cases. That's my two cents. Can you sort them in order of increasing scope?

```suggestion
    it 'registers an offense for `#@`' do
      expect_offense(<<~'RUBY')
        "foo" + '#@bar'
        ^^^^^^^^^^^^^^^ Prefer string interpolation to string concatenation.
      RUBY

      expect_correction(<<~'RUBY')
        "foo\#@bar"
      RUBY
    end

    it 'registers an offense for `#@@`' do
      expect_offense(<<~'RUBY')
        "foo" + '#@@bar'
        ^^^^^^^^^^^^^^^^ Prefer string interpolation to string concatenation.
      RUBY

      expect_correction(<<~'RUBY')
        "foo\#@@bar"
      RUBY
    end

    it 'registers an offense for `#$`' do
      expect_offense(<<~'RUBY')
        "foo" + '#$bar'
        ^^^^^^^^^^^^^^^ Prefer string interpolation to string concatenation.
      RUBY

      expect_correction(<<~'RUBY')
        "foo\#$bar"
      RUBY
    end
```

Reply:
Sure, I moved globals to be last.